src/5BF20.c

```diff
@@ -78,11 +317,30 @@
 
 #pragma GLOBAL_ASM("asm/nonmatchings/5BF20/func_800C0254_5D0F4.s")
 
-#pragma GLOBAL_ASM("asm/nonmatchings/5BF20/func_800C027C_5D11C.s")
-
-#pragma GLOBAL_ASM("asm/nonmatchings/5BF20/func_800C0290_5D130.s")
-
-#pragma GLOBAL_ASM("asm/nonmatchings/5BF20/func_800C02A0_5D140.s")
+s32 func_800C027C_5D11C(void) {
+    s32 temp_v0;
+
+    temp_v0 = D_800E1500_7E3A0;
+    D_800E1500_7E3A0 = 0;
+    return temp_v0;
+}
+
+s32 func_800C0290_5D130(void) {
+    return D_800C21B0_5F050->unk50;
+}
+
+void func_800C02A0_5D140(s32 arg0) {
+    if (arg0 < 0) {
+        arg0 = 0;
+    }
+    if (arg0 >= 7) {
+        arg0 = 6;
+    }
+    D_800C21B0_5F050->unk50 = arg0;
+    if ((D_800C21B0_5F050->unk64 >> 29) < arg0) {
```

Comment:
Have you tested this as a bitfield? Shifting by 29 seems like a _lot_.

Reply:
Yeah, it works with a bitfield. I've added it now.